docs/proof/proof.md

```diff
@@ -0,0 +1,338 @@
+  but for a specific version of the tree.
+
+### Verifying Proofs
+
+The following `RangeProof` methods are used to verify proofs:
+
+* `Verify(rootHash []byte)`: verify that the proof root hash matches the given tree root hash.
+
+* `VerifyItem(key, value []byte)`: verify that the given key exists with the given value, according
+  to the proof.
+
+* `VerifyAbsent(key []byte)`: verify that the given key is absent, according to the proof.
+
+To verify that a `RangeProof` is valid for a given IAVL tree (i.e. that the proof root hash matches
+the tree root hash), run `RangeProof.Verify()` with the tree's root hash:
+
+```go
+// Generate a proof for a=1
+value, proof, err := tree.GetWithProof([]byte("a"))
+if err != nil {
+    log.Fatal(err)
+}
+
+// Verify that the proof's root hash matches the tree's
+err = proof.Verify(tree.Hash())
+if err != nil {
+    log.Fatalf("Invalid proof: %v", err)
+}
+```
+
+The proof must always be verified against the root hash with `Verify()` before attempting other 
+operations. The proof can also be verified manually with `RangeProof.ComputeRootHash()`:
+
+```go
+if !bytes.Equal(proof.ComputeRootHash(), tree.Hash()) {
+    log.Fatal("Proof hash mismatch")
+}
+```
+
+To verify that a key has a given value according to the proof, use `VerifyItem()` on a proof
+generated for this key (or key range):
+
+```go
+// The proof was generated for the item a=1, so this is successful
+err = proof.VerifyItem([]byte("a"), []byte{1})
+if err != nil {
+    log.Printf("Failed to prove a=1: %v", err)
+}
+
+// If we instead claim that a=2, the proof will error
+err = proof.VerifyItem([]byte("a"), []byte{2})
+if err != nil {
+    log.Printf("Failed to prove a=2: %v", err)
+}
+
+// Also, verifying b=2 errors even though it is correct, since the proof is for a=1
+err = proof.VerifyItem([]byte("b"), []byte{2})
+if err != nil {
+    log.Printf("Failed to prove b=2: %v", err)
+}
+```
+
+If we generate a proof for a range of keys, we can use this both to prove the value of any of the 
+keys in the range as well as the absence of any keys that would have been within it:
+
+```go
+// Note that the end key is not inclusive, so c is not in the proof
+keys, values, proof, err := tree.GetRangeWithProof([]byte("a"), []byte("c"))
+if err != nil {
+    log.Fatal(err)
+}
+
+err = proof.Verify(tree.Hash())
+if err != nil {
+    log.Fatal(err)
+}
+
+// Prove that a=1 is in the range
+err = proof.VerifyItem([]byte("a"), []byte{1})
+if err != nil {
+    log.Printf("Failed to prove a=1 with range: %v", err)
+}
+
+// Prove that b=2 is also in the range
+err = proof.VerifyItem([]byte("b"), []byte{2})
+if err != nil {
+    log.Printf("Failed to prove b=2 with range: %v", err)
+}
+
+// Since "ab" is ordered after "a" but before "b", we can prove that it
+// is not in the range and therefore not in the tree at all
+err = proof.VerifyAbsence([]byte("ab"))
+if err != nil {
+    log.Printf("Failed to prove absence of ab: %v", err)
```

Comment:
This is pretty nitpicky, but I think these examples would be a little bit easier to read if it were clearer when we were expecting an error vs. not--every function call is followed by an error check regardless of whether we expect the error or not.

Reply:
Changed it to printing the result, with a comment of the output - is that better?